docs/DeveloperGuide.md

```diff
@@ -2,14 +2,191 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
 
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Add Flashcard Feature
+
+#### Current implementation
+
+The current add flashcard allows the user to add a flashcard to the list of flashcards,
+it is implemented through the following steps:
+
+Step 1:
+The input of user is collected by `getUserCommand()` inside class `Ui`.
+
+Step 2:
+The input string will be converted into a `Command` object by being passed through
+`parseCommand(String userInput)` inside `Parser`.
+
+In this case, an `AddCommand` will be created and returned.
+
+Step 3:
+The `execute()` function of `AddCommand` will run, calling `addNewFlashcard(questionText, answerText)`
+of class `FlashcardList` to create and add the new flashcard to the list.
+
+Then it will also call `printSuccessfulAddMessage(questionText, answerText)` of class `Ui`
+to display text indicating the successful adding function to the user.
+
+At this point, the adding process is completed and the program is ready to take another
+command.
+
+The following sequence diagram show how the add operation works:
+![AddFlashcard-0.png](umlDiagrams%2FAddFlashcard-0.png)
```

Comment:
Perhaps you could distinguish between arrows that return from method calls using dashed arrows.